test/e2e/apibinding/apibinding_permissionclaims_by_name_test.go

```diff
@@ -0,0 +1,467 @@
+	serviceProviderPath := logicalcluster.NewPath(serviceProviderWorkspace.Spec.Cluster)
+	consumerPath := logicalcluster.NewPath(consumerWorkspace.Spec.Cluster)
+
+	t.Logf("provider workspace: %s", serviceProviderWorkspace.Spec.Cluster)
+	t.Logf("consumer workspace: %s", consumerWorkspace.Spec.Cluster)
+
+	cfg := server.BaseConfig(t)
+
+	kcpClusterClient, err := kcpclientset.NewForConfig(cfg)
+	require.NoError(t, err, "failed to construct kcp cluster client for server")
+
+	kubeClusterClient, err := kcpkubernetesclientset.NewForConfig(cfg)
+	require.NoError(t, err, "failed to construct kube cluster client for server")
+
+	t.Logf("installing a sheriff APIResourceSchema and APIExport into workspace %q", serviceProviderPath)
+	apifixtures.CreateSheriffsSchemaAndExport(ctx, t, serviceProviderPath, kcpClusterClient, "wild.wild.west", "board the wanderer")
+
+	sheriffExport := &apisv1alpha1.APIExport{}
+	identityHash := ""
+	framework.Eventually(t, func() (done bool, str string) {
+		sheriffExport, err = kcpClusterClient.Cluster(serviceProviderPath).ApisV1alpha1().APIExports().Get(ctx, "wild.wild.west", metav1.GetOptions{})
+		if err != nil {
+			return false, err.Error()
+		}
+
+		if conditions.IsTrue(sheriffExport, apisv1alpha1.APIExportIdentityValid) {
+			identityHash = sheriffExport.Status.IdentityHash
+			return true, ""
+		}
+		condition := conditions.Get(sheriffExport, apisv1alpha1.APIExportIdentityValid)
+		if condition != nil {
+			return false, fmt.Sprintf("not done waiting for API Export condition status:%v - reason: %v - message: %v", condition.Status, condition.Reason, condition.Message)
+		}
+		return false, "not done waiting for APIExportIdentity to be marked valid, no condition exists"
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "could not wait for APIExport to be valid with identity hash")
+	require.NotNil(t, sheriffExport)
+
+	t.Logf("found identity hash: %v sheriff export", identityHash)
+
+	t.Logf("creating consumer namespace")
+	consumerNS1, err := kubeClusterClient.Cluster(consumerPath).CoreV1().Namespaces().Create(ctx, &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "consumer-ns-1",
+		},
+	}, metav1.CreateOptions{})
+	require.NoError(t, err, "failed to create ns-1")
+
+	t.Logf("waiting for consumer namespace to exist")
+	framework.Eventually(t, func() (done bool, str string) {
+		consumerNS1, err := kubeClusterClient.Cluster(consumerPath).CoreV1().Namespaces().Get(ctx, consumerNS1.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err.Error()
+		}
+
+		if consumerNS1.Status.Phase == v1.NamespaceActive {
+			return true, ""
+		}
+
+		return false, "not done waiting for ns1 to be created"
+	}, wait.ForeverTestTimeout, 110*time.Millisecond, "could not wait for namespace to be ready")
+	t.Logf("namespace %s ready", consumerNS1.Name)
+
+	t.Logf("setting PermissionClaims on APIExport %s", sheriffExport.Name)
+	sheriffExport.Spec.PermissionClaims = makeNarrowCMPermissionClaims("*", "consumer-ns-1")
```

Comment:
@ncdc This permission claim causes an issue with the `kube-root-ca.crt` ConfigMap in the workspace. Specifically, the permission claim cannot be accepted because that ConfigMap doesn't match the criteria. Do we want to make an exception for this configmap? Does that have escalation concerns?

The status of the APIBinding when this fails:

```yaml
status:
  apiExportClusterName: 1wruw3jtfkzt8nhv
  boundResources:
  - group: wild.wild.west
    resource: sheriffs
    schema:
      UID: 8514b50f-221f-4541-831e-22cc9a0dc3f5
      identityHash: 277f31016b73a7ac5cb3e3cf91965d728f5441a66dabe699590996914c614e32
      name: today.sheriffs.wild.wild.west
    storageVersions:
    - v1
  conditions:
  - lastTransitionTime: "2023-03-02T16:49:51Z"
    status: "True"
    type: Ready
  - lastTransitionTime: "2023-03-02T16:49:51Z"
    status: "True"
    type: APIExportValid
  - lastTransitionTime: "2023-03-02T16:49:51Z"
    status: "True"
    type: BindingUpToDate
  - lastTransitionTime: "2023-03-02T16:49:51Z"
    status: "True"
    type: InitialBindingCompleted
  - lastTransitionTime: "2023-03-02T16:49:51Z"
    message: 'Permission claims have not been fully applied: error patching "/v1,
      Resource=configmaps" wrv68fcnqh5ijft4|default/kube-root-ca.crt: configmaps "kube-root-ca.crt"
      is forbidden: <nil>'
    reason: InternalError
    severity: Error
    status: "False"
    type: PermissionClaimsApplied
  - lastTransitionTime: "2023-03-02T16:49:51Z"
    status: "True"
    type: PermissionClaimsValid
  exportPermissionClaims:
  - resource: configmaps
    resourceSelector:
    - names:
      - '*'
      namespaces:
      - consumer-ns-1
```

Reply:
This appears to have been fixed by enabling admission on virtual workspaces and checking permissionclaims there